File: Models/Loan.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookPulse.Models
{
    public class Loan
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [ForeignKey(nameof(Book))]
        public int BookId { get; set; }

        [Required]
        [ForeignKey(nameof(Member))]
        public int MemberId { get; set; }

        [Required]
        public DateTime LoanDate { get; set; } = DateTime.UtcNow;

        public DateTime? ReturnDate { get; set; }

        public DateTime DueDate { get; set; }

        public bool IsReturned { get; set; } = false;

        // Navigation properties
        public virtual Book Book { get; set; } = null!;
        public virtual Member Member { get; set; } = null!;
    }
}

File: frontend/src/types/loan.ts
export interface Loan {
  id: number
  bookId: number
  bookTitle: string
  memberId: number
  memberName: string
  loanDate: string
  returnDate?: string
  dueDate: string
  isReturned: boolean
}

export interface CreateLoanDto {
  bookId: number
  memberId: number
  loanDays?: number
}

export interface ReturnLoanDto {
  loanId: number
}
